test/e2e/unmanaged_dns_test.go

```diff
@@ -0,0 +1,380 @@
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
+	}
+
+	t.Log("Waiting for stable conditions on ingresscontroller after Managed")
+
+	// Wait for the load balancer and DNS to reach stable conditions.
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.ManagedDNS {
+		t.Fatalf("DNSRecord %s expected in ManagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
+	if len(wildcardRecord.Status.Zones) == 0 {
+		t.Fatalf("DNSRecord %s expected allocated dnsZones but found none", wildcardRecordName.Name)
+	}
+
+	verifyUnmanagedIngressController(t, name, "apps."+ic.Spec.Domain, wildcardRecord.Spec.Targets[0])
+}
+
+func TestManagedDNSToUnmanagedDNSIngressController(t *testing.T) {
+	t.Parallel()
+
+	name := types.NamespacedName{Namespace: operatorNamespace, Name: "managed-migrated"}
+	ic := newLoadBalancerController(name, name.Name+"."+dnsConfig.Spec.BaseDomain)
+	ic.Spec.EndpointPublishingStrategy.LoadBalancer = &operatorv1.LoadBalancerStrategy{
+		Scope:               operatorv1.ExternalLoadBalancer,
+		DNSManagementPolicy: operatorv1.ManagedLoadBalancerDNS,
+	}
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller: %v", err)
+	}
+	defer assertIngressControllerDeleted(t, kclient, ic)
+
+	// Wait for the load balancer and DNS to reach stable conditions.
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, availableConditionsForIngressControllerWithLoadBalancer...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	lbService := &corev1.Service{}
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get LoadBalancer service: %v", err)
+	}
+
+	if ingresscontroller.IsServiceInternal(lbService) {
+		t.Fatalf("load balancer %s is internal but should be external", lbService.Name)
+	}
+
+	wildcardRecordName := controller.WildcardDNSRecordName(ic)
+	wildcardRecord := &iov1.DNSRecord{}
+	if err := kclient.Get(context.TODO(), wildcardRecordName, wildcardRecord); err != nil {
+		t.Fatalf("failed to get wildcard dnsrecord %s: %v", wildcardRecordName, err)
+	}
+
+	if wildcardRecord.Spec.DNSManagementPolicy != iov1.ManagedDNS {
+		t.Fatalf("DNSRecord %s expected in ManagedDNS state but got %s", wildcardRecordName.Name, wildcardRecord.Spec.DNSManagementPolicy)
+	}
+
+	verifyUnmanagedIngressController(t, name, "apps."+ic.Spec.Domain, wildcardRecord.Spec.Targets[0])
```

Comment:
(see comments below about `verifyUnmanagedIngressController`). This confuses me a bit. This is a DNS managed ingress controller, but we are calling a function called `verifyUnmanagedIngressController`.

Reply:
Thanks, good catch. This function evolved too much during implementation. I'll rename both the functions (add a godoc as well), 
```
verifyUnmanagedIngressController => verifyExternalIngressController
verifyUnmanagedInternalIngressController => verifyInternalIngressController
```
WDYT?

Reply:
I think that's much clearer